lib/spack/spack/cmd/analyze.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+import llnl.util.tty as tty
+
+import spack.analyzers
+import spack.build_environment
+import spack.cmd
+import spack.cmd.common.arguments as arguments
+import spack.environment as ev
+import spack.fetch_strategy
+import spack.monitor
+import spack.paths
+import spack.report
+
+
+description = "analyze installed packages"
+section = "extensions"
+level = "short"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '--overwrite', action='store_true',
+        help="re-analyze even if the output file already exists.")
+
+    monitor_group = spack.monitor.get_monitor_group(subparser)  # noqa
+
+    subparser.add_argument(
+        '-o', '--outdir', default=None,
+        dest='outdir',
+        help="write output to a different directory than ~/.spack/analyzers")
+    subparser.add_argument(
+        '--list-analyzers', action="store_true",
+        default=False,
+        help="list available analyzers.")
```

Comment:
list and analyze are really different modes here, and everywhere else in spack we use subcommands for this kind fo thing.

I think you want `spack analyze list-analyzers` and `spack analyze run [all|analyzer [...]]`. Also might be good to have `spack analyze list` to show available results (like `spack test`), though that can probably come later.

With subcommands, the options then become less confusing, as `--overwrite` becomes an option on `run` and not on `list-analyzers`, so you can't get weird combinations of args.

Reply:
Good idea - I'll add these as subcommands proper. Let's hold off on adding `list` until we have a good understanding of what we want to show. But I do think I should re-organize analyzer output into it's own namespace, so each analyzer can output whatever content it wants to that directory without worrying about over-writing another analyzer's files.

Reply:
I also think the user might want to run more than one spec, or more than one analyzer, so it makes sense to have as many specs listed as needed, and still take analyzer names with `-a`. E.g.,

```bash
# Run libabigail and diff analyzers on axom and ascent
$ spack analyze run -a libabigail -a diff axom ascent
```

Reply:
Oh I like that a lot